database/oracle/oracle_test.go

```diff
@@ -0,0 +1,216 @@
+package oracle
+
+import (
+	"bytes"
+	"context"
+	"database/sql"
+	"os"
+	"sync"
+	"testing"
+
+	"github.com/golang-migrate/migrate/v4"
+	dt "github.com/golang-migrate/migrate/v4/database/testing"
```

Comment:
Use `github.com/golang-migrate/migrate/v4/dktesting` to test against an Oracle DB running in a Docker container. You can use the postgres tests as an example.

Reply:
Hi, the reason I didn't use docker container for the test cases here are:
1. There are no public official docker images available
1. The oracle image size in the community is 8GB, which is huge
1. The volume size of one single oracle container is about 5GB, which is huge too
1. And more importantly, It will take a long time to start just a single oracle container & configure it(almost 30min on my 16GB memory, 8 cores machine). The test case will run in parallel and each case will require it's own container, which will increase the resource & time costs many times.
 
And Since it's tricky to configure the dependencies for the compile & link automatically because of the oracle download policies, I added oracle build tags into all the oracle related files so that it won't have side effect for the other users who won't need the oracle migration.

Reply:
So basically due to these above reasons, the idea here is the oracle related test cases would be run manually in a pre-configured machine.

Reply:
The image size is less of an issue. But the lack of publicly available docker images is a huge issue.

Also, to avoid environment variable conflicts, use `MIGRATE_TEST_ORACLE_DSN` instead of `ORACLE_DSN`.

Reply:
- MIGRATE_TEST_ORACLE_DSN has changed 
- Now the test case will support running both pre-configured machine and in CI(however I bet there is no such resource to perform the test case in the container)

Reply:
CI test succed